app/controllers/UploadController.scala

```diff
@@ -16,44 +16,47 @@
 
 package controllers
 
-import java.io.{BufferedReader, InputStream, InputStreamReader}
-import java.net.URL
-import java.util.zip.ZipInputStream
-
+import akka.actor.ActorSystem
+import akka.http.scaladsl._
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import akka.stream.scaladsl.Source
 import config.ApplicationConfig
 import controllers.auth.{AuthAction, RequestWithOptionalEmpRef}
-import javax.inject.{Inject, Singleton}
 import models.upscan.{UploadedSuccessfully, UpscanCsvFilesCallbackList}
 import models.{ERSFileProcessingException, SheetErrors}
 import play.api.Logger
 import play.api.i18n.{I18nSupport, Messages}
 import play.api.mvc._
-import services.{CsvFileProcessor, ProcessODSService, StaxProcessor}
-import uk.gov.hmrc.http.HeaderCarrier
+import services.{ProcessCsvService, ProcessODSService, StaxProcessor}
+import uk.gov.hmrc.http.{HeaderCarrier, UpstreamErrorResponse}
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 import utils.ERSUtil
 
-import scala.collection.JavaConverters._
+import java.io.InputStream
+import java.net.URL
+import java.util.zip.ZipInputStream
+import javax.inject.{Inject, Singleton}
 import scala.collection.mutable.ListBuffer
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
 @Singleton
 class UploadController @Inject()(authAction: AuthAction,
                                  processODSService: ProcessODSService,
-                                 csvFileProcessor: CsvFileProcessor,
+                                 csvFileProcessor: ProcessCsvService,
```

Comment:
minor nitpick, if your named the service, could you rename the parameter from `csvFileProcessor` to `processCsvService`

Reply:
sure, will do that along with other review comments